rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/Rename.rsc

```diff
@@ -551,23 +551,42 @@ Tree findCursorInTree(Tree t, loc cursorLoc) {
 list[Tree] extendFocusWithConcreteSyntax([Concrete c, *tail], loc cursorLoc) = [findCursorInTree(c, cursorLoc), c, *tail];
 default list[Tree] extendFocusWithConcreteSyntax(list[Tree] cursor, loc _) = cursor;
 
+@synopsis{
+    Augment the TModel with 'missing' use/def information.
+    Workaround until the typechecker generates this. https://github.com/usethesource/rascal/issues/2172
+}
+TModel augmentTModel(Tree tr, TModel tm, TModel(loc) tmodelForLoc) {
+    tm = augmentFieldUses(tr, tm, tmodelForLoc);
+    tm = augmentFormalUses(tr, tm);
+    return tm;
+}
+
+TModel tmodelForLoc(loc l, PathConfig(loc) getPathConfig)
+    = tmodelForTree(parse(#start[Module], l), getPathConfig);
+
+TModel tmodelForTree(Tree t, PathConfig(loc) getPathConfig) {
+    loc l = t.src.top;
+    pcfg = getPathConfig(l);
+    mname = getModuleName(l, pcfg);
+
+    ccfg = rascalCompilerConfig(pcfg);
+    ms = rascalTModelForNames([mname], ccfg, dummy_compile1);
+
+    <found, tm, ms> = getTModelForModule(mname, ms);
+    if (!found) throw ms.messages;
+    return augmentTModel(t, tm, TModel(loc f) {
+        // Prevent endless recursion
+        if (f == l) return tm;
+        return tmodelForLoc(f, getPathConfig);
```

Comment:
this is now skipping the caching of the rename framework?

Reply:
This is only used when augmenting. That happens outside of the framework. The framework caches the augmented TModel, so this happens once per module at most.